File: Java/src/main/java/org/codecop/lcdnumbers/Scaling.java
package org.codecop.lcdnumbers;

import java.util.Objects;

/**
 * The scale factor value.
 */
public class Scaling {

    public static final Scaling NONE = new Scaling(1);
    public static final Scaling TWO = new Scaling(2);

    private final int times;

    public static Scaling of(int t) {
        return new Scaling(t);
    }

    private Scaling(int times) {
        if (times <= 0) {
            throw new IllegalArgumentException("scaling factor must be >= 1");
        }

        this.times = times;
    }

    public void times(Runnable block) {
        Objects.requireNonNull(block);

        for (int i = 0; i < times; i++) {
            block.run();
        }
    }

    @Override
    public String toString() {
        // debug
        return Integer.toString(times);
    }
}

File: JavaScript/src/scaling.js
exports = typeof window !== 'undefined' && window !== null ? window : global;

/**
 * The scale factor value.
 */
function Scaling(times) {
    'use strict';

    if (times <= 0) {
        throw new RangeError('scaling factor must be >= 1');
    }

    this.none = function() {
        return times === 1;
    };

    this.times = function(block) {
        var i;

        for (i = 0; i < times; i+=1) {
            block();
        }
    };

    this.toString = function() {
        return String(times);
    };

}

exports.Scaling = {};

exports.Scaling.NONE = new Scaling(1);
exports.Scaling.TWO = new Scaling(2);

exports.Scaling.of = function(t) {
    'use strict';
    return new Scaling(t);
};
